synapse/state.py

```diff
@@ -229,6 +236,14 @@ def compute_event_context(self, event, old_state=None):
             else:
                 context.current_state_ids = context.prev_state_ids
 
+            context.state_group = yield self.store.store_state_group(
```

Comment:
do we need to create a new state_group if it's not a state event?

Reply:
Yup. Basically, we've been explicitly told the state at this point (`old_state`) and we need to persist that. This e.g. happens when we're backfilling and we're told the state at this oldest event. We could try and be clever and check if we've already persisted a state group with identical state, but we're not that clever atm.